main.py

```diff
@@ -4,37 +4,337 @@
 
 # Python code for 2D random walk.
 # Source: https://www.geeksforgeeks.org/random-walk-implementation-python/
-import numpy
 import matplotlib.pyplot as plt
+from matplotlib import cm
 import random
+import numpy as np
+import math
+import click
 
-# defining the number of steps
-n = 100000
-
-# creating two array for containing x and y coordinate
-# of size equals to the number of size and filled up with 0's
-x = numpy.zeros(n)
-y = numpy.zeros(n)
-
-# filling the coordinates with random variables
-for i in range(1, n):
-    val = random.randint(1, 4)
-    if val == 1:
-        x[i] = x[i - 1] + 1
-        y[i] = y[i - 1]
-    elif val == 2:
-        x[i] = x[i - 1] - 1
-        y[i] = y[i - 1]
-    elif val == 3:
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] + 1
-    else:
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] - 1
+# user input – click
+_total_steps_option = [
+    click.option(
+        "--total_steps",
+        "-ts",
+        default=10000,
+        type=int,
+        help="Specify the number of total steps for the random walker, Default is 10,000",
+    )
+]
+
+_total_walkers_option = [
```

Comment:
if someone specifies 0 walker, the script is not exited but an empty plot is generated. However, the user get's the error that no handle was found to put in legend. Do you want it like this or should it be aborted?